### PR TITLE
Issue#204

### PR DESCRIPTION
**Pull request by Smaassen2**
Implemented the requested behaviour as mentioned in issue 204.

### CODE REVIEW

File: PANDORA/PMHC/Model.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
 
 class Model:
 
-    def __init__(self, target, model_path='', output_dir = PANDORA.PANDORA_data, pdb=False, molpdf=0, dope=0):
-        '''__init__(self, target, model_path='', output_dir = PANDORA.PANDORA_data, pdb=False, molpdf=0, dope=0)
+    def __init__(self, target, model_path='', output_dir = os.getcwd(), pdb=False, molpdf=0, dope=0):
```
Comment (DarioMarzella) [inline PANDORA/PMHC/Model.py:13]: Please remove the "os.getcwd()" from the function definition. Add an if statement in the beginning of the function instead to assign the output dir if no output dir is provided and specify it in the docstring.

File: PANDORA/PMHC/Model.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
...
+        '''__init__(self, target, model_path='', output_dir = os.getcwd(), pdb=False, molpdf=0, dope=0)
```
Comment (DarioMarzella) [inline PANDORA/PMHC/Model.py:14]: Not that "PANDORA.PANDORA_data is not in the function definition anymore, you can remove this line. Its only role was to preven sphinx from printing the whole PANDORA.PANDORA_data path in the dcoumentation.

File: PANDORA/PMHC/Model.py [hunk 3]
```diff
@@ -10,8 +10,8 @@
 
 class Model:
```
Comment (DarioMarzella) [inline PANDORA/PMHC/Model.py:11]: why does "Model" need an output_dir argument now?

File: PANDORA/PMHC/PMHC.py [hunk 4]
```diff
@@ -258,15 +259,28 @@ def __init__(self, id, peptide, allele_type=[], MHC_class = 'I',
             use_netmhcpan (bool): If True, uses local installation of NetMHCPan to predict the anchors when
                                   anchor positions are not provided. Defaults to False.
             use_templ_seq (bool): If True, if no MHC chain sequences could be retrieved starting from the allele name,
-                                    it will use the best template MHC sequences for the modelling.
-        '''
+                                   it will use the best template MHC sequences for the modelling.
+            output_dir: (string) Path to output directory. Defaults to os.getcwd().
```
Comment (DarioMarzella) [inline PANDORA/PMHC/PMHC.py:263]: please change "Defaults to os.getcwd()." to "Defaults to current working directory"

File: PANDORA/PMHC/PMHC.py [hunk 5]
```diff
@@ -570,3 +584,21 @@ def fill_allele_seq_info(self, use_templ_seq=False):
             except:
                 print('\nWARNING: something went wrong when trying to retrieve chain M allele')
                 print('with blast. Is blastp properly installed as working as "/bin/bash blastp"?')
+
+    def prep_output_dir(self):
```
Comment (DarioMarzella) [inline PANDORA/PMHC/PMHC.py:588]: change this to "make_output_dir"

File: PANDORA/Pandora/Pandora.py [hunk 6]
```diff
@@ -356,18 +312,16 @@ def model(self, n_loop_models=20, n_homology_models=1,
...
-                             logfile=self.logfile)
+                self.__log(self.target.id, 'None', 'Could not find a template')
                 raise Exception('Could not find a template')
 
         print('###############')
         print('TEMPLATE: ', self.template.id)
         # Prepare the output directory
         try:
-            self.prep_output_dir()
+            self.copy_template() #TODO: Use copy_template only here
```
Comment (DarioMarzella) [inline PANDORA/Pandora/Pandora.py:322]: Please remove the "TODO" comment

**Review by DarioMarzella — CHANGES_REQUESTED** [review]

File: PANDORA/PMHC/Model.py (hunk 3 above)
Comment (Smaassen2) [reply PANDORA/PMHC/Model.py:11]: Which part are you referring to? Or do you mean in the whole Model class?

Comment (DarioMarzella) [reply PANDORA/PMHC/Model.py:11]: Oh nevermind. I thought you added it with this PR and I could not see why.

Comment (Smaassen2) [reply PANDORA/PMHC/Model.py:11]: I have removed output_dir from the model class and the code still works. Should I put output_dir back or not?

File: PANDORA/PMHC/PMHC.py [hunk 7]
```diff
@@ -570,3 +584,21 @@ def fill_allele_seq_info(self, use_templ_seq=False):
...
+    def make_output_dir(self):
+        ''' Create an output directory and move the template pdb there
+            Uses self.output_dir (str): Path to output directory. Defaults to os.getcwd().
+        Args:
+            None
+        '''
+
+        # create an output directory
+        try:
+            self.output_dir = '%s/%s' %(self.output_dir, self.id)
```
Comment (DarioMarzella) [inline PANDORA/PMHC/PMHC.py:597]: Please do not redefine the output_dir here, do it directly in the __init__ function and don't change it later.

**Review by DarioMarzella — CHANGES_REQUESTED** [review]

File: PANDORA/Wrapper/Wrapper.py [hunk 8]
```diff
@@ -249,32 +257,48 @@ def __get_targets_from_file(self, data_file, delimiter='\t', header=True,
...
+    def prep_collective_output_dir(self):
+        ''' Create an output directory and move the template pdb there
+            Uses self.output_dir (str): Path to output directory. Defaults to os.getcwd().
+        Args:
+            None
+        '''
+
+        # create an output directory
+        try:
+            generated_id = random.randint(10000, 99999)
```
Comment (DarioMarzella) [inline PANDORA/Wrapper/Wrapper.py:269]: Please do alphanumeric, not only numeric

File: PANDORA/PMHC/PMHC.py (hunk 7 above)
Comment (DarioMarzella) [reply PANDORA/PMHC/PMHC.py:597]: If possible change this to f-string formatting also

**Comment by Smaassen2** [comment]
I fixed the issues in [ea2a1d5](https://github.com/X-lab-3D/PANDORA/pull/209/commits/ea2a1d552e5dae57065eacc0003c88ca7f9e6153)

File: PANDORA/PMHC/Model.py [hunk 9]
```diff
@@ -10,25 +10,27 @@
...
-
         self.target = target
         self.model_path = model_path
         self.molpdf = molpdf
         self.dope = dope
-        self.output_dir = output_dir
+
+        # Define the output directory
+        if output_dir == False:
+            self.output_dir = os.getcwd()
```
Comment (DarioMarzella) [inline PANDORA/PMHC/Model.py:31]: Shouldn't the model object get as default the current working directory + the case directory?

**Review by DarioMarzella — APPROVED** [review]

**Pull request merged**